contracts/Spades.sol

```diff
@@ -113,65 +151,135 @@ contract Spades {
-        return whoSigned[txIndex][_owner];
+
+        return whoSignedTx[txIndex][_owner];
     }
 
-        // Signs a transaction that was submited from another owner
+        // Signs a transaction that was submited from another owner.
     function signTransaction(uint txIndex) public ownerOnly txExists(txIndex){
-        Transaction memory transaction = txMap[txIndex];
-        require(whoSigned[txIndex][msg.sender] == false);
+
+        Transaction storage transaction = txMap[txIndex];
+        require(whoSignedTx[txIndex][msg.sender] == false);
         transaction.confirmations += 1;
-        whoSigned[txIndex][msg.sender] = true;
+        whoSignedTx[txIndex][msg.sender] = true;
         
         emit Sign (msg.sender, txIndex);
     }
 
-        // Revokes a signature that was already made
+        // Revokes a signature that was already made.
     function revokeConfirmation(uint txIndex) public txExists(txIndex) notExecuted(txIndex) ownerOnly {
-        Transaction memory transaction = txMap[txIndex];
 
-        require(whoSigned[txIndex][msg.sender] == true, "You didn't sign this transaction");
+        Transaction storage transaction = txMap[txIndex];
+
+        require(whoSignedTx[txIndex][msg.sender] == true, "You didn't sign this transaction");
         transaction.confirmations - 1;
-        whoSigned[txIndex][msg.sender] = false;
+        whoSignedTx[txIndex][msg.sender] = false;
 
         emit revoked (msg.sender, txIndex);
 
     }
     
-       /// @notice Executes the transaction after reaching the required signatures;
+       /// @notice Executes the transaction passed in submit function after reaching the required signatures;
 
    function executeTransaction(uint txIndex) public txExists(txIndex) {
-        Transaction memory transaction = txMap[txIndex];
+
+        Transaction storage transaction = txMap[txIndex];
         require(transaction.confirmations >= requiredSignatures, "Not enough signatures");
-        (bool success, ) = transaction.to.call{value: transaction.amount}(
+        (bool success, ) = transaction.targetAccount.call{value: transaction.amount}(
             transaction.data);
-        require(success, "Tx failed to execute");
+        require(success, "Tx failed targetAccount execute");
         txMap[txIndex].executed = true;
+        transaction.executed = true;
 
         emit transactionExecuted (msg.sender, txIndex);
    }
+        ///@notice Will submit an account limit targetAccount be signed by the other owners.
+        ///@dev The last withdraw timestamp approved must be fetched for the address submited.
+
+   function submitAccountLimit(address _account, uint _dailyWithdrawlimit) public ownerOnly {
+
+        require(_dailyWithdrawlimit >= 0, "Invalid dailyWithdrawlimit");
+        require(OwnersCheck[_account] = true, "Address does not match with any of the owners.");
+        uint accountTimestamp = accountInfo[_account].lastWithdrawalTimestamp;
```

Comment:
There might not be an entry in that array. If I'm creating a limit for that owner for the first time, this will fail.